src/app/donation-start/donation-start.component.ts

```diff
@@ -203,7 +205,7 @@ export class DonationStartComponent implements AfterContentChecked, AfterContent
           Validators.pattern(emailRegex),
         ]],
         billingCountry: [this.defaultCountryCode], // See setConditionalValidators().
-        billingPostcode: [null],  // See setConditionalValidators().
+        billingPostcode: [null, [Validators.required, ValidateBillingPostCode(this.stripeResponseErrorCode)]],  // See setConditionalValidators().
```

Comment:
@alihejazi I think the fundamental problem here is that you've made a validator-creator, a "thunk", but only the function it returns is relevant after this point. Angular needs a simple validator fn and that's what it's got, which means no further evaluation is ever going to happen. Only the value of `this.stripeResponseErrorCode` _when the form was first created_ can ever be used, so the approach can't work.

It also feels like there is too much Stripe detail inside the validator here anyway – I reckon all we really need is a dumb validator which _always_ says something's wrong when attached.

So with this in mind I would suggest we:
1. simplify the validator to be a tiny fn that always returns an error; and
2. keep the logic to attach or remove it in `DonationStartComponent`, following the logic of or expanding `setConditionalValidators()` to make the validator apply only when we know from Stripe that there is a postcode matching error from the postcode in combination with the given payment card.